### PR TITLE
[DRAFT] detect/uri: Apply bsize constraints, if any

### PR DESCRIPTION
**Pull request by jlucovsky**
Issue: 4226

Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [ ] I have read the contributing guide lines at https://suricata.readthedocs.io/en/latest/devguide/codebase/contributing/contribution-process.html
- [ ] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [ ] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:

Describe changes:
-
-
-

#suricata-verify-pr:
#suricata-verify-repo:
#suricata-verify-branch:
#suricata-update-pr:
#suricata-update-repo:
#suricata-update-branch:
#libhtp-pr:
#libhtp-repo:
#libhtp-branch:

### CODE REVIEW

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW1_stats_chk | | |
| .tcp.rst | 128237 | 103978 | 81.08% |
| .app_layer.error.tls.parser | 931 | 1145 | 122.99% |
| | SURI_TLPR1_stats_chk | | |
| .app_layer.flow.http | 3118951 | 3340795 | 107.11% |
| .app_layer.flow.tls | 510462 | 623355 | 122.12% |
| .app_layer.flow.ssh | 8937 | 11032 | 123.44% |
| .app_layer.flow.telnet | 1624 | 2334 | 143.72% |
| .app_layer.error.ftp-data.gap | 0 | 1 | - |
| | TREX_GENERIC_stats_chk | | |
| .capture.kernel_drops | 0 | 248540 | 0.04 | 

Pipeline 11699

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPR1_stats_chk | | |
| .app_layer.error.ftp-data.gap | 1 | 0 | - | 

Pipeline 12530

File: src/detect-urilen.c [hunk 1]
```diff
@@ -208,6 +208,78 @@ void DetectUrilenApplyToContent(Signature *s, int list)
...
+        if (sm->type != DETECT_CONTENT) {
+            continue;
+        }
+        DetectContentData *cd = (DetectContentData *)sm->ctx;
+        if (cd == NULL) {
+            continue;
+        }
+
+        if (cd->depth == 0 || cd->depth > high) {
+            cd->depth = high;
```
Comment (victorjulien) [inline src/detect-urilen.c:267]: this needs to set `cd->flags |= DETECT_CONTENT_DEPTH;`

**Comment by victorjulien** [comment]
Needs work.

**Pull request closed without merging**